runner/shape.go

```diff
@@ -267,6 +267,11 @@ func shapeMerge(a Shape, b Shape) Shape {
 		Logln(`Missing type equality condition for %s merge: [%#v] and [%#v].`, b.Kind, a, b)
 	}
 
+	// Both scalars of same type; nothing to merge here
+	if a.Kind == ScalarKind && reflect.DeepEqual(a, b) {
```

Comment:
Oh actually I think the reflect.DeepEqual is overkill here. I think it is not as surprising if you keep it as you had it before

Reply:
Oh, right -- thanks for the feedback. I removed the last commit now, it's back to how it was before 👍

Reply:
Perfect! Should be good to merge when tests are passing. Some will fail but just because I can't figure out Github permissions on secrets for non org people.

Thank you so much for the investigation and fix! Fantastic work.